src/formatting_stack/strategies.clj

```diff
@@ -25,25 +28,37 @@
 (speced/defn all-files
   "This strategy unconditionally processes all files."
   [& {:keys [^::protocols.spec/filenames files]}]
-  (let [tracked (->> (impl/file-entries git-command "ls-files")
-                     (impl/absolutize git-command))
-        untracked (->> (impl/file-entries git-command "ls-files" "--others" "--exclude-standard")
-                       (impl/absolutize git-command))]
-    (->> files
-         (into tracked)
-         (into untracked)
-         (impl/extract-clj-files))))
+  ;; This first `binding` is necessary for obtaining an absolutized list of deletions
+  (binding [impl/*skip-existing-files-check?* true]
+    (let [deleted (->> (impl/file-entries git-command "status" "--porcelain")
+                       (filter git-status/deleted-file?)
+                       (map git-status/remove-deletion-markers)
+                       (impl/absolutize git-command)
+                       (set))
+          tracked (->> (impl/file-entries git-command "ls-files")
+                       (impl/absolutize git-command)
+                       (remove deleted))
+          untracked (->> (impl/file-entries git-command "ls-files" "--others" "--exclude-standard")
+                         (impl/absolutize git-command)
+                         (remove deleted))]
+      ;; Second `binding`, to ensure correct results
+      (binding [impl/*skip-existing-files-check?* false]
+        (speced/let [^::impl/existing-files corpus (into tracked untracked)]
```

Comment:
you could remove some duplication by doing this:

```suggestion
          tracked   (impl/file-entries git-command "ls-files")
          untracked (impl/file-entries git-command "ls-files" "--others" "--exclude-standard")]
      ;; Second `binding`, to ensure correct results
      (binding [impl/*skip-existing-files-check?* false]
        (speced/let [^::impl/existing-files corpus (->> (into tracked untracked)
                                                        (impl/absolutize git-command)
                                                        (remove deleted))]
```

Reply:
> you could remove some duplication by doing this:

Will try

Reply:
Cannot be done: the two `(impl/absolutize git-command)` calls must be called within the former binding.

Thankfully, `integration.formatting-stack.strategies` fails if we applied this patch